src/main/scala/com/fulcrumgenomics/bam/SamRecordClipper.scala

```diff
@@ -27,12 +27,32 @@ package com.fulcrumgenomics.bam
 import java.util
 
 import com.fulcrumgenomics.FgBioDef._
+import com.fulcrumgenomics.bam.SamRecordClipper.ClippingMode
 import htsjdk.samtools.{Cigar, CigarElement, SAMRecord, SAMUtils, CigarOperator => Op}
 
 import scala.collection.JavaConversions.iterableAsScalaIterable
 import scala.collection.mutable.ListBuffer
 
 /**
+  * Holds types and constants related to SAMRecord clipping.
+  */
+object SamRecordClipper {
+  /** An enumeration representing the various ways to clip bases within a read. */
+  object ClippingMode extends Enumeration {
+    val Soft, Hard, SoftWithMask = Value
+  }
+
+  type ClippingMode = ClippingMode.Value
+
+  /** The set of tags that should be invalidated if a read undergoes clipping. */
+  val TagsToInvalidate = Seq("MD", "NM", "UQ")
+
+  private val NoCallBase = 'N'.toByte
+  private val NoCallQual = 2.toByte
+}
+
+
```

Comment:
I know there are two newlines here ... but I like the separation between the object and the class 😛